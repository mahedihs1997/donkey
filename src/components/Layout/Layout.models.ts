export interface LayoutProps {
  showLayout: boolean;
  currentRoute: string;
}
