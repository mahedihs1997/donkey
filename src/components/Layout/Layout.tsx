import * as React from 'react';
import Link from 'next/link';
import clsx from 'clsx';
import { useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import MenuIcon from '@material-ui/icons/Menu';
import Home from '@material-ui/icons/Home';
import Face from '@material-ui/icons/Face';
import Polymer from '@material-ui/icons/Polymer';
import Lock from '@material-ui/icons/Lock';

import { LayoutProps } from './Layout.models';
import { useStyles } from './Layout.styles';
import { logout } from '@domain/handlers/logout';

const Layout: React.FunctionComponent<LayoutProps> = (props) => {
  const { children, showLayout } = props;

  const classes = useStyles();
  const theme = useTheme();
  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  if (!showLayout) {
    return <React.Fragment>{children}</React.Fragment>;
  }
  return (
    <div className={classes.root}>
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            className={clsx(classes.menuButton, {
              [classes.hide]: open,
            })}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            The Donkey App
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        variant="permanent"
        className={clsx(classes.drawer, {
          [classes.drawerOpen]: open,
          [classes.drawerClose]: !open,
        })}
        classes={{
          paper: clsx({
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          }),
        }}
      >
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            {theme.direction === 'rtl' ? (
              <ChevronRightIcon />
            ) : (
              <ChevronLeftIcon />
            )}
          </IconButton>
        </div>
        <Divider />
        <List>
          <Link href="/home">
            <ListItem button>
              <ListItemIcon>
                <Home />
              </ListItemIcon>
              <ListItemText primary="Home" />
            </ListItem>
          </Link>
          <Link href="/register">
            <ListItem button>
              <ListItemIcon>
                <Face />
              </ListItemIcon>
              <ListItemText primary="Register" />
            </ListItem>
          </Link>
          <Link href="/login">
            <ListItem button>
              <ListItemIcon>
                <Polymer />
              </ListItemIcon>
              <ListItemText primary="Login" />
            </ListItem>
          </Link>
        </List>
        <Divider />
        <List>
          <ListItem button onClick={logout}>
            <ListItemIcon>
              <Lock />
            </ListItemIcon>
            <ListItemText primary="Logout" />
          </ListItem>
        </List>
      </Drawer>
      <main className={classes.content}>
        <div className={classes.toolbar} />
        {children}
      </main>
    </div>
  );
};

export default Layout;
