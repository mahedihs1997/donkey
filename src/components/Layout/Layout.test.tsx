import React from 'react';
import TestRenderer from 'react-test-renderer';

import Layout from '.';

describe('<Layout />', () => {
  it('should wrap children with the layout', () => {
    const testRenderer = TestRenderer.create(
      <Layout showLayout={true} currentRoute="/home">
        <div>Hi</div>
      </Layout>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('should render children without the layout', () => {
    const testRenderer = TestRenderer.create(
      <Layout showLayout={false} currentRoute="/home">
        <div>Hi</div>
      </Layout>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
