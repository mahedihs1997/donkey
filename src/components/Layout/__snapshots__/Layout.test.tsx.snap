// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render children without the layout 1`] = `
<div>
  Hi
</div>
`;

exports[`<Layout /> should wrap children with the layout 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-3 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-bar-2 MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        Donkey - 
      </h6>
      <div
        className="MuiAvatar-root MuiAvatar-circle"
      >
        <img
          alt="Image of something"
          className="MuiAvatar-img"
          src="https://png.pngtree.com/png-clipart/20190705/original/pngtree-cartoon-european-and-american-character-avatar-design-png-image_4366075.jpg"
        />
      </div>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6 makeStyles-drawerClose-8"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerClose-8 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="makeStyles-toolbar-9"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Home
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Users
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Jobs
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4v2h16V4zm1 10v-2l-1-5H4l-1 5v2h1v6h10v-6h4v6h2v-6h1zm-9 4H6v-4h6v4z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Parts
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Payments
            </span>
          </div>
        </div>
      </ul>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-colorError"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Logout
            </span>
          </div>
        </div>
      </ul>
    </div>
  </div>
  <main
    className="makeStyles-content-10"
  >
    <div
      className="makeStyles-toolbar-9"
    />
    <div>
      Hi
    </div>
  </main>
</div>
`;
